### PR TITLE
Including RuntimePackAsset in DesignerRuntimeImplementationProjectOutputGroup

### PR DESCRIPTION
**Pull request by Shyam-Gupta**
`RuntimePackAsset` paths are not available in SDK and they are required by WinForms .NET designer.

### CODE REVIEW

**Review by dsplaisted — APPROVED** [review]
`RuntimePackAsset` items are generally in a Shared Framework and can be loaded by the designers without shadow copying.  The exception to this is the Microsoft.Windows.SDK.NET.Ref runtime pack.  Those items can be identified by checking the `RuntimePackAlwaysCopyLocal` metadata on them.  If that is set to true, then they should be included in the set of files to shadow copy.  If it's not set to true, then they should continue to be excluded.

It would be nice to have test coverage of this change.  There is one test that covers the shadow copying: `GivenThatWeWantDesignerSupport.It_provides_runtime_configuration_and_shadow_copy_files_via_outputgroup`.  That test could be expanded or a similar one added to cover the Windows SDK reference.

Also, this PR is currently targeting main, which will be .NET 9.  You probably want to target it to 8.0.2xx (or 8.0.1xx if you want to take the fix through servicing).

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets [hunk 1]
```diff
@@ -128,12 +128,18 @@ Copyright (c) .NET Foundation. All rights reserved.
...
       <!--
-        For .NET Core, we do not include NuGet package assets or runtime pack
-        assets, irrespective of self-contained / copy-local settings. Designer
+        For .NET Core, we do not include NuGet package assets,
+        irrespective of self-contained / copy-local settings. Designer
         will load these from the NuGet cache or shared framework always.
       -->
       <_DesignerShadowCopy
-        Remove="@(_ResolvedCopyLocalBuildAssets);@(RuntimePackAsset)"
+        Remove="@(_ResolvedCopyLocalBuildAssets)"
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets:136]: At this point the `_DesignerShadowCopy` list includes all of the runtime pack assets and the files from NuGet packages, and we want to remove those (except the runtime packs in the case of RuntimePackAlwaysCopyLocal).  I think what you want is something like this (assuming I got the syntax right):

```xml
      <!-- For .NET Core, we do not include NuGet package assets, as the designer will load these from the NuGet cache. -->
      <_DesignerShadowCopy
        Remove="@(_ResolvedCopyLocalBuildAssets)"
        Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'"
        />

      <!-- For .NET Core runtime pack assets, we do not include them regardless of whether the app is
              self-contained, as they will be loaded from the shared framework.  However, for runtime pack
              assets where RuntimePackAlwaysCopyLocal is true, there is no shared framework, so we still
              include those.  -->
      <_DesignerShadowCopy
        Remove="@(RuntimePackAsset)"
        Condition="'%(RuntimePackAsset.RuntimePackAlwaysCopyLocal)' != 'true'"
        />

```

**Review by dsplaisted — COMMENTED** [review]
Since we're modifying logic that interacts with self-contained, it might be good to update the test to have some cases where the app tested is self-contained.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets (hunk 1 above)
Comment (Shyam-Gupta) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DesignerSupport.targets:132]: Thanks, I have updated this file as per your suggestion and confirmed that its working fine. I have also updated the test to run for `SelfContained` apps.
I remember testing `_DesignerShadowCopy` at that point earlier but found that it was not containing `RuntimePackAsset` paths. But I think I made some mistake in my testing.

Let me know if you have more feedback. Thanks.

File: src/Tests/Microsoft.NET.Build.Tests/GiventThatWeWantDesignerSupport.cs [hunk 2]
```diff
@@ -14,12 +14,18 @@ public GivenThatWeWantDesignerSupport(ITestOutputHelper log) : base(log)
         }
 
         [Theory]
-        [InlineData("net46")]
-        [InlineData("netcoreapp3.0")]
-        [InlineData("net5.0-windows")]
-        public void It_provides_runtime_configuration_and_shadow_copy_files_via_outputgroup(string targetFramework)
+        [InlineData("net46", "true")]
+        [InlineData("net46", "false")]
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GiventThatWeWantDesignerSupport.cs:18]: Self-contained is something that only applies to .NET Core, so we can remove a case here.

```suggestion
        [InlineData("net46", "false")]
```

**Review by dsplaisted — APPROVED** [review]
Looks good.  Feel free to merge when ready.

**Pull request merged**